### ISSUE TITLE
"Upload Lambda": does not check assumptions on the uploaded directory is invalid

### ISSUE THREAD

**Issue by justinmk3** [aws/aws-toolkit-vscode#2567]
### Problem

The `Upload Lambda` function expects various directory contents to exist, such as `template.yaml`. If you try `Upload Lambda` on an invalid directory it fails with an obscure null reference message(s):

<img width="533" alt="image" src="https://user-images.githubusercontent.com/55561878/162279739-d9218a08-e04e-4417-8b60-3d80b947c1d1.png">

### Expected behavior

1. Check expected directory contents, or try/catch whatever is failing 
2. show a useful error message.

**Comment by JadenSimon** [comment]
I'm guessing this only occurs with image-based lambdas?
Could not repro with the zip ones.

**Comment by justinmk3** [comment]
can reproduce by giving an invalid value to a template property, e.g. `Timeout={too big}`:

```
Resources:
    HelloWorldFunction:
        Type: AWS::Serverless::Function
        Properties:
            CodeUri: hello_world/
            Handler: app.lambda_handlerPy38
            Runtime: python3.8
            Timeout: 99999
```

**Comment by justinmk3** [comment]
Fixed in AWS Toolkit 1.63
